net/tcp/tcp_input.c

```diff
@@ -933,22 +923,22 @@ static void tcp_input(FAR struct net_driver_s *dev, uint8_t domain,
                         continue;
                       }
                     else if (opt == TCP_OPT_MSS &&
-                              dev->d_buf[hdrlen + 1 + i] == TCP_OPT_MSS_LEN)
+                             IPDATA(tcpiplen + 1 + i) == TCP_OPT_MSS_LEN)
                       {
                         uint16_t tcp_mss = TCP_MSS(dev, iplen);
 
                         /* An MSS option with the right option length. */
 
                         tmp16 =
-                          (dev->d_buf[hdrlen + 2 + i] << 8) |
-                          dev->d_buf[hdrlen + 3 + i];
+                          (IPDATA(tcpiplen + 2 + i) << 8) |
+                          IPDATA(tcpiplen + 3 + i);
```

Comment:
can be combined to two lines instead of 3

Reply:
Done!